sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -4483,6 +4478,17 @@ class Dataset[T] private[sql](
     }
   }
 
+  /** Returns a optimized plan for CommandResult, convert to `LocalRelation`. */
+  private def withCommandResultOptimized: Dataset[T] = {
```

Comment:
`with` prefix is for a wrapper. Should name it differently, e.g., `commandResultOptimized`.